library/src/main/java/com/novoda/downloadmanager/Batch.java

```diff
@@ -1,30 +1,34 @@
 package com.novoda.downloadmanager;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
 
-public final class Batch {
+public class Batch {
 
     private final DownloadBatchId downloadBatchId;
     private final String title;
-    private final Map<DownloadFileId, NetworkAddressAndFilePath> networkAddressAndFilePathById;
+    private final List<File> files;
 
-    Batch(DownloadBatchId downloadBatchId, String title, Map<DownloadFileId, NetworkAddressAndFilePath> networkAddressAndFilePathById) {
+    public static Builder with(DownloadBatchId downloadBatchId, String title) {
+        return new LiteBatchBuilder(downloadBatchId, title, new ArrayList<>());
+    }
+
+    Batch(DownloadBatchId downloadBatchId, String title, List<File> files) {
         this.downloadBatchId = downloadBatchId;
         this.title = title;
-        this.networkAddressAndFilePathById = networkAddressAndFilePathById;
+        this.files = files;
     }
 
-    DownloadBatchId getDownloadBatchId() {
+    public DownloadBatchId downloadBatchId() {
         return downloadBatchId;
     }
 
-    String getTitle() {
+    public String title() {
```

Comment:
Just a question: Is this the getter naming we have for the `v2` of the library? Is there any discussion link around this? I personally prefer regular getters because it is more usable in kotlin and in java, they're ok.

Reply:
There hasn't been a conversation around this, I think most on the project prefer without the `get`

Reply:
You mean our internal projects or `download-manager`?

Reply:
`download-manager`

Reply:
Consistency ftw 👍